acceptance/chaos_test.go

```diff
@@ -107,33 +116,47 @@ func TestChaos(t *testing.T) {
 		defer close(teardown)
 		rnd, seed := randutil.NewPseudoRand()
 		log.Warningf("monkey starts (seed %d)", seed)
-		for round := 1; time.Now().Before(deadline); round++ {
+		for atomic.StoreInt64(&round, 1); time.Now().Before(deadline); atomic.AddInt64(&round, 1) {
+			curRound := atomic.LoadInt64(&round)
 			select {
 			case <-stopper:
 				return
 			default:
 			}
 			nodes := rnd.Perm(num)[:rnd.Intn(num)+1]
-
-			log.Infof("round %d: restarting nodes %v", round, nodes)
+			log.Infof("round %d: restarting nodes %v", curRound, nodes)
 			for i := 0; i < num; i++ {
 				clients[i].Lock()
 			}
-			for i := 0; i < num; i++ {
-				log.Infof("restarting %v", i)
+			for _, i := range nodes {
+				// Two early exit conditions.
+				select {
+				case <-stopper:
+					break
```

Comment:
I don't see how.
1) all clients are locked
2) selection of clients are restarted
3) all clients are unlocked

The break will only break the inner loop of 2, so all clients will unlock.
But I can remove the early exit conditions since the kill/restart/init isn't stopping the test from shutting down, the puts are.